zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbResourceState.java

```diff
@@ -0,0 +1,303 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.state.deployment;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import io.camunda.zeebe.db.ColumnFamily;
+import io.camunda.zeebe.db.TransactionContext;
+import io.camunda.zeebe.db.ZeebeDb;
+import io.camunda.zeebe.db.impl.DbCompositeKey;
+import io.camunda.zeebe.db.impl.DbForeignKey;
+import io.camunda.zeebe.db.impl.DbLong;
+import io.camunda.zeebe.db.impl.DbString;
+import io.camunda.zeebe.db.impl.DbTenantAwareKey;
+import io.camunda.zeebe.db.impl.DbTenantAwareKey.PlacementType;
+import io.camunda.zeebe.engine.EngineConfiguration;
+import io.camunda.zeebe.engine.state.mutable.MutableResourceState;
+import io.camunda.zeebe.protocol.ZbColumnFamilies;
+import io.camunda.zeebe.protocol.impl.record.value.deployment.ResourceRecord;
+import java.util.Optional;
+import org.agrona.DirectBuffer;
+
+public class DbResourceState implements MutableResourceState {
+
+  private static final int DEFAULT_VERSION_VALUE = 0;
+
+  private final DbString tenantIdKey;
+  private final DbLong dbResourceKey;
+  private final DbTenantAwareKey<DbLong> tenantAwareResourceKey;
+  private final DbForeignKey<DbTenantAwareKey<DbLong>> fkResourceKey;
+  private final PersistedResource dbPersistedResource;
+  private final ColumnFamily<DbTenantAwareKey<DbLong>, PersistedResource> resourcesByKey;
+  private final DbString dbResourceId;
+  private final VersionManager versionManager;
+
+  private final DbLong resourceVersion;
+  private final DbCompositeKey<DbString, DbLong> idAndVersionKey;
+  private final DbTenantAwareKey<DbCompositeKey<DbString, DbLong>> tenantAwareIdAndVersionKey;
+  private final ColumnFamily<DbTenantAwareKey<DbCompositeKey<DbString, DbLong>>, PersistedResource>
```

Comment:
❌ This is a bad idea, we shouldn't duplicate a persisted resource, we already store it once in the `resourcesByKey` CF. I can see that this was done as well in the `DbFormState` and it looks like an oversight.

It's better to use the `fkResourceKey` as the value here. You can check how it's done in the `DbDecisionState` [here](https://github.com/camunda/camunda/blob/e4ed843b4f1909f0a585c8d19c898d4030828d69/zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbDecisionState.java#L78-L80).